src/electionguard/api/tally_votes.c

```diff
@@ -143,7 +136,7 @@ bool initialize_trustees(struct trustee_state *trustee_states)
         if (result.status != DECRYPTION_TRUSTEE_SUCCESS)
             ok = false;
         else
-            trustees[i] = result.decryptor;
+            decryption_trustees[result.trustee_index] = result.decryptor;
```

Comment:
since most of the internals of this lib expect things to be index based and in the right order, by assigning the trustee to it's expected spot in the `decryption_trustees` array, we side-step having to do lookups in other parts of the code